### PR TITLE
feat: add feature to control secure_channel

### PR DESCRIPTION
**Pull request by Xynnn007**
For non-confidential container scenarios, `secure_channel` is not supported. This commit brings a feature `getresource` to control whether to enable `secure_channel`.

Also, add flexibly configurable for paths to support both confidential scenarios and non-confidential scenarios.
Close https://github.com/confidential-containers/image-rs/issues/110

Signed-off-by: Xynnn007 <xynnn@linux.alibaba.com>

### CODE REVIEW

**Comment by Xynnn007** [comment]
The eaakbc test fails because the test container environment is not up-to-date, let me fix this.

**Comment by Xynnn007** [comment]
related pr https://github.com/confidential-containers/image-rs/pull/114

**Comment by Xynnn007** [comment]
cc @jiangliu @arronwy

File: src/image.rs [hunk 1]
```diff
@@ -183,11 +179,12 @@ impl ImageClient {
 
         // If one of self.config.auth and self.config.security_validate is enabled,
         // there will establish a secure channel between image-rs and Attestation-Agent
+        #[cfg(feature = "getresource")]
```
Comment (jiangliu) [inline src/image.rs:182]: Seems it may help readability by grouping code between line 182-263 as a helper function?

Comment (Xynnn007) [reply src/image.rs:182]: As `secure_channel` of line 182 will also be used in line 289, and between them there should be other irrelevant code. If a function, there should be two different functions. One of them return a tuple `(SecureChannel, Auth)`, and the other return `Auth` controlled by feature, which will be more confusing. I also have not thought of a better way to refactor : (

**Review by jiangliu — APPROVED** [review]

**Review by jialez0 — APPROVED** [review]

**Pull request merged**